crypto/x509/x509_vfy.c

```diff
@@ -3101,7 +3113,7 @@ static int build_chain(X509_STORE_CTX *ctx)
 
                 /*
                  * We've added a new trusted certificate to the chain, recheck
-                 * trust.  If not done, and not self-signed look deeper.
+                 * trust.  If not done, and not self-issued look deeper.
```

Comment:
Self-signednes is in general too strong as an indication to stop looking deeper while building a cert chain: there are cases where a candidate cert for inclusion in the chain is just self-issued but not self-signed, for instance `test/certs/ee-ed25519.pem`. If the chain building algorithm did not stop already on an self-issued cert it would start looping at this point since the issuer name of the cert is by definition the same as its subject name.

Reply:
I guess you mean "Self-issued is in general ..."

Reply:
But your second sentence isn't making sense to me either. If it's self-issued but not self-signed, and not trusted, I expect it to look for something with the same subject that is different than the current certificate.

Reply:
Where the trusted part is only when partial_chain is set.

Reply:
> I guess you mean "Self-issued is in general ..."

No, as self-signed is (strictly) stronger than self-issued: it implies self-issued but not the other way round.

Reply:
> If it's self-issued but not self-signed, and not trusted, I expect it to look for something with the same subject that is different than the current certificate.

Right, but the aspect of chain building you refer to here is implemented elsewhere.